packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -89,23 +78,84 @@ export function removeCertificatesWithoutIssuer(ca: string[]): {
     }
     return { pem, parsed };
   });
-  caWithParsedCerts = caWithParsedCerts.filter(({ parsed }) => {
-    const keep =
-      !parsed ||
-      parsed.checkIssued(parsed) ||
-      caWithParsedCerts.find(
-        ({ parsed: issuer }) => issuer && parsed.checkIssued(issuer)
-      );
-    if (!keep) {
-      messages.push(
+}
+
+function doesCertificateHasMatchingIssuer(
+  { parsed }: ParsedX509Cert,
+  certs: ParsedX509Cert[]
+) {
+  return (
+    !parsed ||
```

Comment:
So if parsed is falsey we return that it does have a matching issuer?

Reply:
I suppose that's old code.

Reply:
Yes, I'm just keeping it as-is, but FWIW [Annas reasoning](https://github.com/mongodb-js/devtools-shared/pull/467#discussion_r1745678311) in the PR that introduced it seems fair to me. I'll add it as a comment here so that we don't need to track down the PR every time

Reply:
Ended up moving the `!parsed` check outside of the method just so this function name clearly describes and matches what it does. Moved the not parsed check to the filter itself and added a comment, seems a bit cleaner that way?